src/Symfony/Component/HttpClient/DataCollector/HttpClientDataCollector.php

```diff
@@ -194,7 +194,7 @@ private function getCurlCommand(array $trace): ?string
         $dataArg = [];
 
         if ($json = $trace['options']['json'] ?? null) {
-            $dataArg[] = '--data '.escapeshellarg(json_encode($json, \JSON_PRETTY_PRINT));
+            $dataArg[] = '--data '.escapeshellarg(self::jsonEncode($json, \JSON_PRETTY_PRINT));
```

Comment:
Why not just use self::jsonEncode without any flag? I don't see the need to pass pretty print the data - also because this is NOT what was sent so this won't replay exactly the same request.

Reply:
Is there any risk of diff between pretty and dirty JSON? Anyway, as it's only for the copy-as-curl, I'm ok to remove the pretty flag :+1: